x-pack/plugins/canvas/public/components/es_index_select/index.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+import usePrevious from 'react-use/lib/usePrevious';
+import { getIndices } from '../../lib/es_service';
+import { ESIndexSelect as Component, ESIndexSelectProps as Props } from './es_index_select';
+
+type ESIndexSelectProps = Omit<Props, 'indices' | 'loading'>;
+
+export const ESIndexSelect: React.FunctionComponent<ESIndexSelectProps> = (props) => {
+  const { value, onChange } = props;
+  const prevValue = usePrevious(value);
+
+  const [indices, setIndices] = useState<string[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (prevValue !== value) {
```

Comment:
Prior to your changes, the index loading was in a `componentDidMount` so it only would make the request once.  Now it's in a hook that's dependent on the selected value, so it reruns on change.  Should probably change that.

Reply:
@crob611, I've updated the code, based on your nits.